Assets/MRTK/Providers/XRSDK/Controllers/GenericXRSDKController.cs

```diff
@@ -263,29 +271,7 @@ protected virtual void UpdateSingleAxisData(MixedRealityInteractionMapping inter
                     {
                         CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
                     }
-                }
-
-                switch (interactionMapping.InputType)
```

Comment:
I think collapsing these two checks together might lead to unexpected behavior. For example, if the float data is changing from 0.1 to 0.2, we'll now go into the first (now only) `interactionMapping.Changed` case, which then checks if `interactionMapping.BoolData` is true, which sends `OnInputUp` if not. So we'll probably get a whole bunch of `OnInputUp` events on our way to a full press (and the same on the way back to fully unpressed).

Reply:
ohhhhh I see why the original design was that way, I'll revert it and add comments explaining why